src/Validator/AuthorCommitteeValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace AppBundle\Validator;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Committee;
+use AppBundle\Repository\CommitteeMembershipRepository;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @Annotation
+ */
+class AuthorCommitteeValidator extends ConstraintValidator
+{
+    private $committeeMembershipRepository;
+    private $security;
+
+    public function __construct(
+        CommitteeMembershipRepository $committeeMembershipRepository,
+        Security $security
+    ) {
+        $this->committeeMembershipRepository = $committeeMembershipRepository;
+        $this->security = $security;
+    }
+
+    /**
+     * @param Committee $committee
+     * @param AuthorCommittee $constraint
+     */
+    public function validate($committee, Constraint $constraint)
+    {
+        if (!$constraint instanceof AuthorCommittee) {
+            throw new UnexpectedTypeException($constraint, AuthorCommittee::class);
+        }
+
+        if (null === $committee) {
+            return;
+        }
+
+        if (!$committee instanceof Committee) {
+            throw new UnexpectedTypeException($committee, Committee::class);
+        }
+
+        $adherent = $this->security->getUser();
+        if(!$adherent instanceof Adherent) {
+            throw new UnexpectedTypeException($adherent, Adherent::class);
+        }
+
+        if (!$this->committeeMembershipRepository->isAdherentInCommittee($adherent, $committee)) {
```

Comment:
you can do 
```php
if (null === $user->getMembershipFor($committee)) {
  // ...
}
```

Reply:
this will trigger a query fetching all committees...
i prefer using a query only counting what i need

what do you think ?

Reply:
oh yes, good point 👍